### PR TITLE
[Gazebo 9] Added Profiler to gazebo::rendering

### PR DESCRIPTION
**Pull request by ahcorde**
Added profiler to gazebo::rendering

Selección_053

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by chapulina** [comment]
It looks like a compiler warning slipped in with #2813 , we could fix it now:

```
RemoteryConfig.h:8
"RMT_USE_OPENGL" redefined #define RMT_USE_OPENGL 1
```

**Comment by chapulina** [comment]
This PR is behind #2837 . Once that goes in, this one should be updated.

**Comment by ahcorde** [comment]
@chapulina updated with the last changes in Gazebo 11 PR

**Comment by j-rivero** [comment]
@osrf-jenkins run tests please

File: gazebo/sensors/CameraSensor.cc [hunk 1]
```diff
@@ -248,8 +251,9 @@ void CameraSensor::Render()
   }
   else
   {
-    if (!this->camera || !this->IsActive() || !this->NeedsUpdate())
+    if (!this->camera || !this->IsActive() || !this->NeedsUpdate()){
```
Comment (chapulina) [inline gazebo/sensors/CameraSensor.cc]: oops

File: gazebo/sensors/MultiCameraSensor.cc [hunk 2]
```diff
@@ -294,6 +295,7 @@ void MultiCameraSensor::Render()
     std::lock_guard<std::mutex> lock(this->dataPtr->cameraMutex);
     if (this->dataPtr->cameras.empty())
     {
+      GZ_PROFILE_END();
```
Comment (chapulina) [inline gazebo/sensors/MultiCameraSensor.cc]: These `END`s can be removed

File: gazebo/sensors/GpuRaySensor.cc [hunk 3]
```diff
@@ -603,19 +603,23 @@ void GpuRaySensor::PrerenderEnded()
...
+    }
 
     this->dataPtr->laserCam->Render();
     this->dataPtr->rendered = true;
     this->dataPtr->renderNeeded = false;
   }
   else
   {
-    if (!this->dataPtr->laserCam || !this->IsActive() || !this->NeedsUpdate())
+    if (!this->dataPtr->laserCam || !this->IsActive() || !this->NeedsUpdate()){
```
Comment (chapulina) [inline gazebo/sensors/GpuRaySensor.cc]: another style oops

File: gazebo/rendering/skyx/src/VCloudsManager.cpp [hunk 4]
```diff
@@ -3,7 +3,7 @@
 This source file is part of SkyX.
 Visit http://www.paradise-studios.net/products/skyx/
 
-Copyright (C) 2009-2012 Xavier Vergu�n Gonz�lez <xavyiy@gmail.com>
+Copyright (C) 2009-2012 Xavier Vergu�n Gonz�lez <xavyiy@gmail.com>
```
Comment (chapulina) [inline gazebo/rendering/skyx/src/VCloudsManager.cpp]: Not a big deal, but it's interesting that this happened here

**Review by chapulina — APPROVED** [review]
I just have some minor comments which we can fix before merging. CI is running and should be done in a couple hours.

**Comment by chapulina** [comment]
> It looks like a compiler warning slipped in with #2813 , we could fix it now:

My fix on ab3094c worked locally, but the warning is still on CI. Since it's not introduced by this PR, I don't think it should prevent merging it.

**Pull request merged**